### PR TITLE
KAFKA-15842: Correct handling of KafkaConsumer.committed for new consumer

### PR DESCRIPTION
**Pull request by AndrewJSchofield**
This PR fixes some details of the interface to KafkaConsumer.committed which were different between the existing consumer and the new consumer.

Adds a unit test that validates the behaviour is the same for both consumer implementations.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by AndrewJSchofield** [comment]
Closing and reopening PR to retry failed build.

**Comment by AndrewJSchofield** [comment]
https://issues.apache.org/jira/browse/KAFKA-15932 opened for flaky test.

File: checkstyle/suppressions.xml [hunk 1]
```diff
@@ -48,7 +48,7 @@
     <suppress id="dontUseSystemExit"
               files="Exit.java"/>
     <suppress checks="ClassFanOutComplexity"
-              files="(AbstractFetch|Sender|SenderTest|ConsumerCoordinator|KafkaConsumer|PrototypeAsyncConsumer|KafkaProducer|Utils|TransactionManager|TransactionManagerTest|KafkaAdminClient|NetworkClient|Admin|KafkaRaftClient|KafkaRaftClientTest|RaftClientTestContext).java"/>
```
Comment (kirktrue) [inline checkstyle/suppressions.xml:51]: Thank you for this clean up!

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java [hunk 2]
```diff
@@ -663,6 +662,10 @@ public Map<TopicPartition, OffsetAndMetadata> committed(final Set<TopicPartition
                     time.timer(timeout));
                 committedOffsets.forEach(this::updateLastSeenEpochIfNewer);
                 return committedOffsets;
+            } catch (TimeoutException e) {
+                throw new TimeoutException("Timeout of " + timeout.toMillis() + "ms expired before the last " +
+                    "committed offset for partitions " + partitions + " could be determined. Try tuning default.api.timeout.ms " +
```
Comment (kirktrue) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java:667]: nit: please replace the literal string `default.api.timeout.ms` with the appropriate constant from `ConsumerConfig` 🥺

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java [hunk 3]
```diff
@@ -653,7 +652,7 @@ public Map<TopicPartition, OffsetAndMetadata> committed(final Set<TopicPartition
         try {
             maybeThrowInvalidGroupIdException();
             if (partitions.isEmpty()) {
-                return new HashMap<>();
+                return Collections.emptyMap();
```
Comment (kirktrue) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java:655]: Does this make a difference, or just a stylistic/pet peeve/optimiz/sation thing?

File: core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala [hunk 4]
```diff
@@ -594,8 +594,9 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     assertThrows(classOf[InvalidTopicException], () => consumer.partitionsFor(";3# ads,{234"))
   }
 
+  // Temporarily do not run flaky test for consumer group protocol
   @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
-  @MethodSource(Array("getTestQuorumAndGroupProtocolParametersAll"))
+  @MethodSource(Array("getTestQuorumAndGroupProtocolParametersGenericGroupProtocolOnly"))
```
Comment (kirktrue) [inline core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala:599]: 😭

**Review by kirktrue — CHANGES_REQUESTED** [review]
Thanks @AndrewJSchofield!

Thanks for the fix and the associated clean up. One minor nit-ty change requested.

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java (hunk 3 above)
Comment (AndrewJSchofield) [reply clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java:655]: I made it strictly identical to the previous implementation.

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java (hunk 2 above)
Comment (cadonna) [reply clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java:667]: While I agree with @kirktrue , let's merge this PR and fix this on the side in a follow-up PR.

**Review by cadonna — APPROVED** [review]
Thanks @AndrewJSchofield !

LGTM!

**Pull request merged**